### PR TITLE
Add Context.IsReplaying E2E tests

### LINKED ISSUE #3332: Add E2E tests covering context.IsReplaying() across all languages/backends
**Is your feature request related to a problem? Please describe.**
None of the E2E tests cover context.IsReplaying or equivalent, and an issue was discovered with Python+DTS where this behavior is not working correctly. Add tests to prevent future regressions in this area

**Are there any existing GitHub discussions or issues filed that help give some context to this proposal?**
https://github.com/Azure/azure-functions-durable-python/issues/595

**Describe the solution you'd like**
New E2E tests covering this behavior

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by andystaples**
# Summary
## What changed?
- Adds E2E tests that verify the behavior of orchestration context.IsReplaying
- Also makes an improvement across the E2E tests to use polling-based log detection

## Why is this change needed?
- Bug found in Python, see linked work item 

## Issues / work items
- Resolves #3332 
- Related #

---

# Project checklist
- [x] Documentation changes are not required
  - [ ] Otherwise: Documentation PR is ready to merge and referenced in `pending_docs.md`
- [x] Release notes are not required for the next release
  - [ ] Otherwise: Notes added to `release_notes.md`
- [x] Backport is not required
  - [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
- [x] All required tests have been added/updated (unit tests, E2E tests)
- [x] No extra work is required to be leveraged by OutOfProc SDKs
  - [ ] Otherwise: Work tracked here: #issue_or_pr_in_each_sdk
- [x] No change to the version of the `WebJobs.Extensions.DurableTask` package
  - [ ] Otherwise: Major/minor updates are reflected in `/src/Worker.Extensions.DurableTask/AssemblyInfo.cs`
- [x] No EventIds were added to `EventSource` logs
  - [ ] Otherwise: Ensure EventIds are within the supported range in the existing Windows infrastructure (validate via deployed telemetry). If needed, extend the range via a PR such as https://msazure.visualstudio.com/One/_git/AAPT-Antares-Websites/pullrequest/7463263?_a=files
- [ ] This change should be added to the `v2.x` branch
  - [x] Otherwise: This change applies exclusively to `WebJobs.Extensions.DurableTask` v3.x and will be retained only in the `dev` and `main` branches
- [ ] Breaking change?
  - [ ] If yes:
    - Impact:
    - Migration guidance:
---

# AI-assisted code disclosure (required)
## Was an AI tool used? (select one)
- [ ] No
- [ ] Yes, AI helped write parts of this PR (e.g., GitHub Copilot)
- [x] Yes, an AI agent generated most of this PR

If AI was used:
- Tool(s): VS Copilot
- AI-assisted areas/files: All
- What you changed after AI output: Fixed test skip attributes and small syntactic errors

AI verification (required if AI was used):
- [x] I understand the code and can explain it
- [x] I verified referenced APIs/types exist and are correct
- [x] I reviewed edge cases/failure paths (timeouts, retries, cancellation, exceptions)
- [x] I reviewed concurrency/async behavior
- [x] I checked for unintended breaking or behavior changes

---

# Testing
## Automated tests
- Result: Passed / Failed (link logs if failed)

## Manual validation (only if runtime/behavior changed)
- Environment (OS, .NET version, components):
- Steps + observed results:
  1.
  2.
  3.
- Evidence (optional):

---

# Notes for reviewers
- N/A

### CODE REVIEW

File: test/e2e/Tests/Tests/IsReplayingTests.cs
Comment (bachuv) [inline test/e2e/Tests/Tests/IsReplayingTests.cs:1]: Can you add some comments explaining what each test should do? For example, expected behavior and output?

Comment (andystaples) [reply test/e2e/Tests/Tests/IsReplayingTests.cs:1]: Done

File: test/e2e/Tests/Tests/IsReplayingTests.cs [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        JsonNode outputJson = JsonNode.Parse(details.Output)!;
+
+        // On the final pass, only the code after the activity (non-replaying)
+        // increments the live-log counter.
+        Assert.Equal(1, outputJson["live_log_count"]!.GetValue<int>());
+        Assert.Equal("logged", outputJson["activity_result"]!.GetValue<string>());
+
+        await Task.Delay(2000);
+        string logs = string.Join(Environment.NewLine, this.fixture.TestLogs.CoreToolsLogs);
+        Assert.Contains("IsReplayingConditionalLog: LIVE after activity", logs);
```
[the remark this answers is not in the document]
Comment (andystaples) [reply test/e2e/Tests/Tests/IsReplayingTests.cs]: Updated all log access to use polling-based log detection

File: test/e2e/Tests/Helpers/TestLoggerProvider.cs [hunk 2]
```diff
@@ -21,6 +22,29 @@ public TestLoggerProvider(IMessageSink messageSink)
...
+            {
+                return;
+            }
+
+            await Task.Delay(250);
+        }
+
+        // Final check after deadline.
+        Assert.True(this.logs.Any(predicate), failureMessage ?? $"Expected log was not found within {maxWaitSeconds}s timeout.");
+    }
```
[the remark this answers is not in the document]
Comment (andystaples) [reply test/e2e/Tests/Helpers/TestLoggerProvider.cs:61]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/Azure/azure-functions-durable-extension/pull/3356#discussion_r2885323846)

**Review by bachuv — APPROVED** [review]
